semver/semver.go

```diff
@@ -0,0 +1,117 @@
+package semver
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	RegexSemver     = `^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?(\+([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?$`
+	RegexPreRelease = `^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)`
+)
+
+type SemVer struct {
+	major      int
+	minor      int
+	patch      int
+	preRelease string
+	build      string
+}
+
+func ParseSemver(input string) (SemVer, error) {
+	parsed, err := parseSemver(input)
+	return parsed, err
+}
+
+// CompareSemver returns three values -1, 0, +1
+// -1 denotes ver1 < ver2
+// 0 denotes invalid input
+// +1 denotes ver1 > ver2
+func CompareSemver(ver1, ver2 string) (int, error) {
```

Comment:
Change this to Compare as the package name will already be used while invoking . semver.Compare(<>)

Reply:
Lets have 2 functions

1.func CompareRaw(ver1, ver2 string) (int, error) 

 2. func (s *Semver) Compare(v *semver) (int,error)